### DISCUSSION TITLE
How do I add weeks?

### DISCUSSION THREAD

**Discussion by ghost** [CodenameCrew/CodenameEngine#218] [Q&A] [answered]
Self-explanatory. I didn't find any YT tutorials or stuff on the wiki on how to actually implement a week in game. Do you have to actually hardcode it in the game or something?

**Comment by charlesisfeline** [comment, marked as answer]
you can, in xml format (btw put it in assets/weeks or mods/[insert mod name here]/data/weeks) (we are not kade engine!)
```xml
<!DOCTYPE codename-engine-week>
<week name="DADDY DEAREST" chars="dad,bf,gf" sprite="week1">
	<song>Bopeebo</song>
	<song>Fresh</song>
	<song>Dadbattle</song>
</week>
```
what im showing here, is an example of a Codename week metadata file.


or you can just put a song in the mods folder (``mods/[insert mod name here/songs/``), which will automatically appear on the freeplay menu, just in case you wanna add some freeplay songs or smth

hope this works out ;)

**Comment by ryauxryo-svg** [comment]
how to add specific difficulty for the week like in mod that only use difficulty hard?

**Reply by r6915ee** [reply]
Add the `difficulty` tag in the file inside of the week tag like the rest of the songs:

```xml
<difficulty name="Hard"/>
```
